File: Example/app/src/main/java/com/stripe/example/TerminalEventListener.java
package com.stripe.example;

import com.google.common.collect.Lists;
import com.stripe.stripeterminal.ConnectionStatus;
import com.stripe.stripeterminal.PaymentStatus;
import com.stripe.stripeterminal.Reader;
import com.stripe.stripeterminal.ReaderEvent;
import com.stripe.stripeterminal.TerminalListener;

import java.util.List;

/**
 * The {@code TerminalEventListener} implements the {@link TerminalListener} interface and will
 * forward along any events to other parts of the app that register for updates.
 */
public class TerminalEventListener implements TerminalListener {

    private static List<TerminalListener> listeners = Lists.newArrayList();

    private static ConnectionStatus currentConnectionStatus = ConnectionStatus.NOT_CONNECTED;
    private static PaymentStatus currentPaymentStatus = PaymentStatus.NOT_READY;

    static void registerListener(TerminalListener listener) {
        listeners.add(listener);
    }

    @Override
    public void onReportReaderEvent(ReaderEvent event) {
        for (TerminalListener listener : listeners) {
            listener.onReportReaderEvent(event);
        }
    }

    @Override
    public void onReportLowBatteryWarning() {
        for (TerminalListener listener : listeners) {
            listener.onReportLowBatteryWarning();
        }
    }

    @Override
    public void onUnexpectedReaderDisconnect(Reader reader) {
        for (TerminalListener listener : listeners) {
            listener.onUnexpectedReaderDisconnect(reader);
        }
    }

    @Override
    public void onConnectionStatusChange(ConnectionStatus status) {
        currentConnectionStatus = status;
        for (TerminalListener listener : listeners) {
            listener.onConnectionStatusChange(status);
        }
    }

    @Override
    public void onPaymentStatusChange(PaymentStatus status) {
        currentPaymentStatus = status;
        for (TerminalListener listener : listeners) {
            listener.onPaymentStatusChange(status);
        }
    }

    /**
     * @return the current ConnectionStatus
     */
    public static ConnectionStatus getCurrentConnectionStatus() {
        return currentConnectionStatus;
    }

    /**
     * @return the current PaymentStatus
     */
    public static PaymentStatus getCurrentPaymentStatus() {
        return currentPaymentStatus;
    }
}

File: Example/app/src/main/java/com/stripe/example/TerminalEventListener.kt
package com.stripe.example

import android.util.Log
import com.stripe.stripeterminal.*

/**
 * The `TerminalEventListener` implements the [TerminalListener] interface and will
 * forward along any events to other parts of the app that register for updates.
 *
 * TODO: Finish implementing
 */
class TerminalEventListener : TerminalListener {

    override fun onReportReaderEvent(event: ReaderEvent) {
        Log.i("ReaderEvent", event.toString())
    }

    override fun onReportLowBatteryWarning() {
        Log.i("LowBatteryWarning", "")
    }

    override fun onUnexpectedReaderDisconnect(reader: Reader) {
        Log.i("UnexpectedDisconnect", reader.serialNumber)
    }

    override fun onConnectionStatusChange(status: ConnectionStatus) {
        Log.i("ConnectionStatusChange", status.toString())
    }

    override fun onPaymentStatusChange(status: PaymentStatus) {
        Log.i("PaymentStatusChange", status.toString())
    }
}
